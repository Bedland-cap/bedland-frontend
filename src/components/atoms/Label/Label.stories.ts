import type { Meta, StoryObj } from '@storybook/react';
import Label from './Label';

const meta = {
  title: 'atoms/Label',
  component: Label,
  tags: ['autodocs'],
} satisfies Meta<typeof Label>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    label: 'login',
    htmlFor: 'login',
    textColor: 'white',
    size: 'small',
  },
};
