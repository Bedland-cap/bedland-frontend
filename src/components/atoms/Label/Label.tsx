import Lbl from './Label.styled';

interface ILabel {
  htmlFor: string;
  label: string;
  textColor: string;
  size: 'small' | 'medium' | 'large';
}

const Label = ({ htmlFor, label, textColor, size }: ILabel) => (
  <Lbl htmlFor={htmlFor} textColor={textColor} size={size}>
    {label}
  </Lbl>
);

export default Label;
