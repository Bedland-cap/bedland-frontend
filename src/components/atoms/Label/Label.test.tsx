import renderer from 'react-test-renderer';
import Label from './Label';

describe('Label', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(<Label label='login' htmlFor='login' textColor='white' size='small' />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
