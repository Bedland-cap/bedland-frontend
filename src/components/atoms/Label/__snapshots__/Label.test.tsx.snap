// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<label
  className="sc-bcXHqe EwzGf"
  htmlFor="login"
  size="small"
>
  login
</label>
`;
